### PR TITLE
Adding Reviung39 shield

### PR DESCRIPTION
**Pull request by jshreynolds**
I've added the Reviung39 shield to the repository and updated all of the relevant scripts and documentation. 

The firmware built is running and manually tested on my Reviung39.   

Happy to make any fixes or changes upon request!

### CODE REVIEW

**Comment by innovaker** [comment]
Thanks for the contribution @jshreynolds!

Please could you replace the merge with a rebase?  Due to current limitations in the setup scripts, the PR will probably need to be rebased after it's been through review, so removing the merge now will make that simpler.

**Comment by jshreynolds** [comment]
@innovaker rebased!  and thank y'all for all the work you're doing on this fantastic project!

**Comment by innovaker** [comment]
Much easier, thanks!  We've a queue for the shields at the moment so it might take a while to get through, but we'll loop back around to this PR eventually.  Thanks in anticipation of your patience!

>thank y'all for all the work you're doing on this fantastic project!

🙇‍♂️

File: app/boards/shields/reviung39/reviung39.conf
Comment (Nicell) [inline app/boards/shields/reviung39/reviung39.conf:6]: This should probably be removed as it's not a board specific option. We should be setting sane defaults for `ZMK_SLEEP` as well.

File: docs/static/setup.sh
Comment (Nicell) [inline docs/static/setup.sh:94]: I believe this should be added on to the end rather than right before the Reviung41.

**Review by Nicell — CHANGES_REQUESTED** [review]
@jshreynolds Thanks for your work on this shield! Looks very familiar. Just a few comments on some tiny changes. Otherwise, it looks good!

File: app/boards/shields/reviung39/reviung39.conf
Comment (jshreynolds) [reply app/boards/shields/reviung39/reviung39.conf]: Better late than never!  Done!

File: docs/static/setup.sh
Comment (jshreynolds) [reply docs/static/setup.sh]: Right! Missed that correlation with the below line. Fixed!

**Comment by jshreynolds** [comment]
Howdy, @Nicell !  I've made the fixes suggested and took the liberty of rebasing against the latest main branch.  I hope that was in line with your workflow!  Should be ready to merge, but let me know if you need any other changes.  I won't let so much time elapse next time!

**Comment by petejohanson** [comment]
Thank you, contributor, for your patience with how long review and merge of boards/shields has taken!

There are three recent refactors/changes to boards and shields that require some attention, and then we can finally get this PR merged!

1. Hardware Metadata
1. Pro Micro shield DT naming changes
1. Split changes for BLE advertising

# Hardware Metadata

## The Problem

When first developing the process around contributing new shields/boards to ZMK, we failed to recognize that several key files (setup scripts, documentation page of supported hardware, and GH Action `build.yml` file) required changes, often in the same spot, for every PR. This resulted in immediate merge conflicts for every *other* PR after one was merged, which is a  headache for contributors.

## The Fix

By adding discrete metadata files that are located with the boards/shields in question, and using that metadata to generate setup scripts, website hardware list, etc., users can contributing new hardware descriptions without the need to change the same files that other contributors are changing.

## Next Steps

First, refer to https://zmk.dev/docs/development/hardware-metadata-files to familiarize yourself with the new metadata file format.

Next, you have two options for fixing up your PR:

1. If comfortable with `git rebase`, perform an interactive rebase and revert any changes to `build.yml`, `hardware.md`, or the setup scripts `setup.sh`/`setup.ps1`, and then add the new metadata YAML file. Then force push your branch. Or,
2. Create a new branch from an up-to-date `main`, copy in the files for your new hardware, add the metadata file, then commit and push the new branch. Then, edit your open PR to point to your new branch.

# Pro Micro shield DT naming changes

In #876, we have simplified the DT naming for the "nexus node" we expose for pro-micro compatible boards, deprecating the use of `pro_micro_a`, and renaming `pro_micro_d` node to simply `pro_micro`. For pro-micro boards and shields, you'll need to adjust your DT to use the proper names.

Please see https://zmk.dev/docs/development/new-shield#shield-overlays for the updated docs on this.

# Split Shield Advertising Changes

In addition, if this is a split PR, please see #658 where we have changed our conventions to *remove* the the name from the right sides, to prevent users attempting to pair with them and causing split sync issues. This also includes removing the " Left" suffix from the naming on the left side. See the changes in that PR for examples of what to change with your split shield.

# Getting Help

If you have any questions about any of these changes, please comment here and tag @zmkfirmware/boards-shields or ask in the `#boards-shields` Discord channel.

**Comment by Nicell** [comment]
Hello! We're closing this PR due to there being no activity since the metadata update to ZMK in September of last year. If you are still interested in updating this PR, please let us know, and we can reopen it and review your changes (hopefully a lot faster this time)!

**Pull request closed without merging**
